src/services/extractVehicleBooking.ts

```diff
@@ -0,0 +1,79 @@
+import { ITestResult } from '../interfaces/ITestResult';
+import { IBooking } from '../interfaces/IBooking';
+import { DynamoDBStreamEvent } from 'aws-lambda';
+import DynamoDB from 'aws-sdk/clients/dynamodb';
+import logger from '../util/logger';
+
+const trimTestCode = (testCode: string | undefined): string => {
+  if (testCode === undefined) throw new Error('testCode not defined in test result');
+
+  if (testCode.length === 4) {
+    if (+testCode[3] !== +testCode[3]) throw new Error(`4th char of test code is non-numeric: ${testCode}`);
+
+    return testCode.slice(0, 3);
+  } 
+
+  return testCode;
+};
+
+const trimTestStationName = (testStationName: string): string => {
+  if (testStationName.length > 10) return testStationName.slice(0, 10);
+  
+  return testStationName;
+};
+
+export const extractVehicleBookings = (event: DynamoDBStreamEvent): IBooking[] => {
+  let bookings: IBooking[] = [];
+
+  for (const dbRecord of event.Records) {
+    if (dbRecord.eventName !== 'INSERT') {
+      logger.info(`${dbRecord.eventName} event - ignoring`);
+      continue;
+    }
+
+    if (!dbRecord.dynamodb?.NewImage) continue;
+    
+    const newImage = DynamoDB.Converter.unmarshall(dbRecord.dynamodb.NewImage) as ITestResult;
+    
+    if (newImage.testResultId.toLowerCase().includes('legacy')) {
+      logger.info('legacy test result - ignoring');
+      continue;
+    }
+        
+    bookings = [...bookings, ...extractBookingDetails(newImage)];
+  }
+
+  return bookings;
+};
+
+export const extractBookingDetails = (testResult: ITestResult): IBooking[] => {
+  try {
+    return testResult.testTypes.map((testType) => {
+      if (testResult.vehicleType === 'trl') {
+        if (testResult.trailerId) return {
+          name: trimTestStationName(testResult.testStationName),
+          bookingDate: testResult.testStartTimestamp as string,
+          vrm: testResult.trailerId,
+          testCode: trimTestCode(testType.testCode),
+          testDate: testResult.testStartTimestamp as string,
+          pNumber: testResult.testStationPNumber,
+        };
+        throw new Error('Trailer does not have trailerId available');
+      }
+      
+      if (testResult.vrm) return {
+        name: trimTestStationName(testResult.testStationName),
+        bookingDate: testResult.testStartTimestamp as string,
+        vrm: testResult.vrm,
+        testCode: trimTestCode(testType.testCode),
+        testDate: testResult.testStartTimestamp as string,
+        pNumber: testResult.testStationPNumber,
+      };
+
+      throw new Error(`${testResult.vehicleType} does not have associated vrm`);
+    });
+  } catch (error) {
+    logger.error(error);
```

Comment:
From my experience you need to use this overload to get the error details. Just having the Error parameter logs 'undefined'.
```suggestion
    logger.error('Error', error);
```

Reply:
For this instance there won't be an error stack associated with the error so I think it's fine just to have `logger.error(error)`, I'd say to have it as the latter if the error was caused from something external like AWS.

Reply:
It's the way Winston behaves. If you are logging an Error, it needs to be in the second parameter:
![image](https://user-images.githubusercontent.com/13391709/188708623-3f22f1bc-c78b-4253-9b51-9154888d8132.png)